src/main/java/uk/gov/mca/beacons/api/services/DeleteBeaconService.java

```diff
@@ -1,23 +1,60 @@
 package uk.gov.mca.beacons.api.services;
 
-import java.util.UUID;
+import static java.lang.String.format;
+import static java.time.LocalDateTime.now;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import uk.gov.mca.beacons.api.domain.AccountHolder;
+import uk.gov.mca.beacons.api.domain.Note;
+import uk.gov.mca.beacons.api.domain.NoteType;
+import uk.gov.mca.beacons.api.dto.DeleteBeaconRequestDTO;
+import uk.gov.mca.beacons.api.exceptions.AccountHolderNotFoundException;
+import uk.gov.mca.beacons.api.gateways.AccountHolderGateway;
 import uk.gov.mca.beacons.api.gateways.BeaconGateway;
+import uk.gov.mca.beacons.api.gateways.NoteGateway;
 
 @Service
 @Transactional
 public class DeleteBeaconService {
 
+  private static final String TEMPLATE_REASON_TEXT =
+    "The account holder deleted the record with reason: '%s'";
+
   private final BeaconGateway beaconGateway;
+  private final AccountHolderGateway accountHolderGateway;
+  private final NoteGateway noteGateway;
 
   @Autowired
-  public DeleteBeaconService(BeaconGateway beaconGateway) {
+  public DeleteBeaconService(
+    BeaconGateway beaconGateway,
+    AccountHolderGateway accountHolderGateway,
+    NoteGateway noteGateway
+  ) {
     this.beaconGateway = beaconGateway;
+    this.accountHolderGateway = accountHolderGateway;
+    this.noteGateway = noteGateway;
   }
```

Comment:
Is this open to extension if a member of the beacon registry team needs to delete a beacon?

Reply:
Again, as above.  I wasn't entirely sure how we would handle that scenario so made the assumption that it would only be an account holder making the request to delete a beacon so went with injecting the `AccountHolderGateway`.

The problem, that doesn't sit quite well with me, is that for a back office user we grab the currently logged in user through the `AuthGateway`, where we are extracting the claims from the JWT for things like `name`, `email`, `authId`. To make this more generic to handle a registry team member to delete a beacon as well as an account holder would require us to have a gateway which could get a generic `User` object.  What didn't sit well was conditionally having to get a user by some id vs extracting it from the JWT claims where we wouldn't use that ID.


Worth a discussion tomorrow morning!  I was a little hesitant to try and abstract/generalise the user concept too early incase it causes more overhead than it was worth vs just specifying that it is an account holders id who is issuing the delete request for a beacon.